jujugui/static/gui/src/test/test_app.js

```diff
@@ -1056,6 +1056,136 @@ describe('File drag over notification system', function() {
     });
 
   });
+  describe('switchEnv', function() {
+    var Y, app, container;
+    var _generateMockedApp = function(sandbox, socketUrl) {
+      app = new Y.juju.App({
+        container: container,
+        viewContainer: container,
+        sandbox: sandbox,
+        consoleEnabled: true,
+        user: 'admin',
+        password: 'admin',
+        jujuCoreVersion: '1.21.1.1-trusty-amd64',
+        charmstorestore: new window.jujulib.charmstore(),
+        sandboxSocketURL: 'ws://host:port/ws/environment/undefined/api'
+      });
+      var fake_ws = {
+        onclose: function() { this.oncloseCalled = true; },
+        oncloseCalled: false
+      };
+      var fake_ecs = {
+        clear: function() { this.clearCalled = true; },
+        clearCalled: false
+      }; 
+      var fake_env = {
+        ecs: fake_ecs,
+        ws: fake_ws,
+        closeCalled: false,
+        socketUrl: socketUrl || 'wss://example.com/ws',
+        setUser: 'not-called',
+        setPassword: 'not-called',
+        close: function() { this.closeCalled = true; },
+        get: function(key) {
+          if (key === 'socket_url') {
+            return this.socketUrl; 
+          }
+          if (key === 'ecs') {
+            return this.ecs;
+          }
+        },
+        set: function(key, val) {
+          if (key === 'socket_url') {
+            this.socketUrl = val;
+          } 
+        },
+        setCredentials: function(obj) {
+          this.setUser = obj.user;
+          this.setPassword = obj.password;
+        }
+      };
+      var fake_db = {
+        resetCalled: false,
+        fireSignal: null,
+        reset: function() { this.resetCalled = true; },
+        fire: function(signal) { this.fireSignal = signal; }
+      };
+      app.env = fake_env;
+      app.db = fake_db;
+      return app;
+    }; 
+
+    before(function(done) {
+      Y = YUI(GlobalConfig).use(['juju-gui'], function(Y) {
+        done();
+      });
+    });
+
+    beforeEach(function() {
+      container = Y.Node.create('<div id="test" class="container"></div>');
+    });
+
+    afterEach(function() {
+      if (app) {
+        app.destroy({remove: true});
+      }
+    });
+
+    it('does not switch envs in sandbox', function() {
```

Comment:
Thanks for adding tests for switchEnv! And sorry that you didn't find them already: the switchEnv was initially intended as demo code, thanks for making it production ready code.